Feature/data context namespace

jcampbell opened the pull request:
This PR:
  1. Bring the latest changes from develop into feature/data_context
  2. Adds a new generator GlobReaderGenerator that can accept glob patterns to build data_assets, with associated tests
  3. Refactors develop changes into the new reader API
  4. Updates documentation for new API

cselig committed:
remove separate spark tests

jcampbell committed:
Merge pull request #485 from cselig/consolidate-tests

Remove separate spark tests

jcampbell committed:
Use more specific exception handling in _to_match_json_schema

jcampbell committed:
Improve multi-dialect sqlalchemy regex handling

jcampbell committed:
Fix typo in json_schema patch

jcampbell committed:
Merge pull request #489 from great-expectations/fix/jsonschema

Fix/jsonschema

Aylr committed:
* improve wording on expectation grid

Aylr committed:
* updated links and doc name

jcampbell committed:
Merge pull request #498 from Aylr/aylr/tiny-doc-clarification

* improve wording on expectation grid

abegong committed:
Update README.md

jcampbell committed:
Merge pull request #499 from great-expectations/abegong-patch-1

Update README.md

miguel-komodo committed:
Add spark parquet context

miguel-komodo committed:
Add test for spark csv data context

miguel-komodo committed:
Add test for parquet

miguel-komodo committed:
Fix typo with spark parquet data context

miguel-komodo committed:
Update docs

miguel-komodo committed:
Merge branch 'develop' into feature/parquet-context

jcampbell committed:
Merge pull request #500 from mastratton3/feature/parquet-context

Feature/parquet context

jcampbell committed:
Merge remote-tracking branch 'origin/develop' into feature/data_context_namespace

 -- Fully fixing this merge will involve adapting the parquet context to the new
    datasource / generator format

jcampbell committed:
Add support for known file extensions/readers and glob-based filesystem generator

jcampbell committed:
Update docstrings

jcampbell committed:
Remove unicode character; add encoding info

jcampbell committed:
Remove unicode characters in comment; add coding note

jcampbell committed:
py2 compatibility affordances

jcampbell committed:
Error handling and defaults improvements

jcampbell committed:
Revise cli language

eugmandel commented on docs/source/data_contexts.rst:
```diff
@@ -22,6 +22,7 @@ There are currently four types of data contexts:
   - :ref:`PandasCSVDataContext`: The PandasCSVDataContext ('PandasCSV') exposes a local directory containing files as datasets.
   - :ref:`SqlAlchemyDataContext`: The SqlAlchemyDataContext ('SqlAlchemy') exposes tables from a SQL-compliant database as datasets.
   - :ref:`SparkCSVDataContext`: The SparkCSVDataContext ('SparkCSV') exposes csv files accessible from a SparkSQL context.
+  - :ref:`SparkParquetDataContext`: The SparkParquetDataContext ('SparkParquet') exposes parquet files accessible from a SparkSQL context.
```
Aren't the old *DataContext classes are going away? How should we deal with deprecation?

jcampbell replied:
Yes; I think we should document it, and view that as part of the comprehensive documentation update (but I wanted to keep all the docs in sync with bringing in develop first)

eugmandel commented on great_expectations/data_context/data_context.py:
```diff
@@ -39,11 +41,37 @@
+      - The datasource actually connects to a source of materialized data and returns Great Expectations
+        DataAssets connected to a compute environment and ready for validation.
+
+      - The Generator knows how to intropsect datasources and produce identifying "batch_kwargs" that define
```
typo: intropsect -> introspect

jcampbell replied:
Good catch; will fix

jcampbell committed:
Update run_id default; add get_validation_result for context

jcampbell commented:
I just added two new features into this PR:
1. Change autogenerated run_id from UUID to ISO8601 UTC timestamp at now()
2. Added preliminary support for get_validation_result (but it's brittle around run_ids)

I've also started adding content to the roadmap/changelog and will continue updating that.

eugmandel commented on great_expectations/data_context/data_context.py:
```diff
@@ -953,12 +981,11 @@ def profile_datasource(self, datasource_name, generator_name=None, profiler=Basi
-            except Exception as exc:
-                logger.warning("\tSomething went wrong when profiling %s. (Perhaps a loading error?) Skipping." % (name))
+            except IOError as exc:
+                logger.warning("\IOError while profiling %s. (Perhaps a loading error?) Skipping." % (name))
```
Are we letting all other errors propagate all the way up?

jcampbell replied:
I'm not sure yet what other errors might be...but before the behavior was to eat *any* error, which I think of as generally bad practice so I figured we'd let them go up if such errors exist and develop a testing plan once we know what other errors might be

jcampbell committed:
Add note about migrating between versions

jcampbell committed:
Add support for get_validation_result; update root_directory

jcampbell committed:
Add prototype for sqlalchemy type checking

jcampbell committed:
Update typo

jcampbell committed:
Improve support for sqlalchemy types

jcampbell committed:
Update type_list tests

jcampbell committed:
Initial update of tests for mysql support

jcampbell committed:
Minor CLI init usability tweaks

jcampbell committed:
Add mysqlclient to requirements-dev

jcampbell committed:
Correct sqlite schema definition

jcampbell committed:
Use stricter type names in test definition schemas

jcampbell committed:
Remove infinity column for mysql support

abegong commented:
Is this mergeable?

jcampbell committed:
Disable WIP mysql tests

jcampbell committed:
Re-enable infinities column existence test
